naga/src/back/spv/block.rs

```diff
@@ -253,6 +258,445 @@ impl Writer {
+
+            let vert_to_copy_ptr = self.id_gen.next();
+            body.push(Instruction::access_chain(
+                self.get_pointer_type_id(
+                    return_info.vertex_type_id,
+                    spirv::StorageClass::Workgroup,
+                ),
+                vert_to_copy_ptr,
+                vert_array_ptr,
+                &[val_i],
+            ));
+
+            // Load the entire vertex value
+            let vert_to_copy = self.id_gen.next();
+            body.push(Instruction::load(
+                return_info.vertex_type_id,
+                vert_to_copy,
+                vert_to_copy_ptr,
+                None,
+            ));
+
+            let mut builtin_index = 0;
+            let mut binding_index = 0;
+            // Write individual members of the vertex
+            for (member_id, member) in return_info.vertex_members.iter().enumerate() {
+                let val_to_copy = self.id_gen.next();
+                let mut needs_y_flip = false;
+                body.push(Instruction::composite_extract(
+                    member.ty_id,
+                    val_to_copy,
+                    vert_to_copy,
+                    &[member_id as u32],
+                ));
+                let ptr_to_copy_to = self.id_gen.next();
+                // Get a pointer to the struct member to copy
+                match member.binding {
+                    crate::Binding::BuiltIn(bi) => {
+                        body.push(Instruction::access_chain(
+                            self.get_pointer_type_id(member.ty_id, spirv::StorageClass::Output),
+                            ptr_to_copy_to,
+                            return_info.vertex_builtin_block.unwrap(),
+                            &[
+                                val_i,
+                                self.get_constant_scalar(crate::Literal::U32(builtin_index)),
+                            ],
+                        ));
+                        needs_y_flip = matches!(bi, crate::BuiltIn::Position { .. })
+                            && self.flags.contains(WriterFlags::ADJUST_COORDINATE_SPACE);
+                        builtin_index += 1;
+                    }
+                    crate::Binding::Location { .. } => {
+                        body.push(Instruction::access_chain(
+                            self.get_pointer_type_id(member.ty_id, spirv::StorageClass::Output),
+                            ptr_to_copy_to,
+                            return_info.vertex_bindings[binding_index],
+                            &[val_i, zero_u32],
+                        ));
+                        binding_index += 1;
+                    }
+                }
+                body.push(Instruction::store(ptr_to_copy_to, val_to_copy, None));
+                // Flip the vertex position y coordinate in some cases
+                // Can't use epilogue flip because can't read from this storage class I believe
```

Comment:
```suggestion
                // Can't use epilogue flip because can't read from this storage class
```

Try not to use `I` in commentary.